core/src/main/java/org/mule/util/NotificationUtils.java

```diff
@@ -94,13 +96,16 @@ public static void addMessageProcessorPathElements(List<MessageProcessor> proces
                 MessageProcessorPathElement messageProcessorPathElement;
 
                 // To avoid adding a level in some path elements:
-                if (!(mp instanceof InterceptingChainLifecycleWrapperPathSkip))
+                if (mp instanceof InterceptingChainLifecycleWrapperPathSkip || mp instanceof TransactionalInterceptingMessageProcessor)
                 {
-                    messageProcessorPathElement = parentElement.addChild(mp);
+                    //This messageProcessorPathElement is fictional and is never added to the path but is needed to continue adding children recursively
+                    messageProcessorPathElement = new DefaultMessageProcessorPathElement(mp,String.valueOf(0)); //The name could be any since it's never added to any path
```

Comment:
Can you move this logic to AbstractInterceptingMessageProcessorBase#addMessageProcessorPathElements?